.ci/jenkins/Jenkinsfile.buildchain

```diff
@@ -0,0 +1,110 @@
+@Library('jenkins-pipeline-shared-libraries')_
+
+additionalTimeout = "${env.ADDITIONAL_TIMEOUT?.trim() ?: 120}"
+
+agentLabel = "${env.ADDITIONAL_LABEL?.trim() ? ADDITIONAL_LABEL : 'kie-rhel7 && kie-mem16g'} && !master"
+build_jdk_tool = env.BUILD_JDK_TOOL?.trim() ?: 'kie-jdk11'
+build_maven_tool = env.BUILD_MAVEN_TOOL?.trim() ?: 'kie-maven-3.8.1'
+
+pr_type = env.PR_TYPE?.trim() ?: 'pr'
+
+pipeline {
+    agent {
+        label agentLabel
+    }
+    tools {
+        nodejs 'nodejs-12.16.3'
+    }
+    options {
+        timestamps ()
+        timeout(time: additionalTimeout, unit: 'MINUTES')
+    }
+    environment {
+        FIREFOX_FOLDER = '/opt/tools/firefox-60esr'
+    }
+    stages {
+        stage('Initialize') {
+            steps {
+                sh 'printenv'
+            }
+        }
+        // executes a script that compresses the consoleText and attaches it to the mail
+        stage('build sh script') {
+            steps {
+                script {
+                    mailer.buildLogScriptPR()
+                }
+            }
+        }
+        stage('check space before build') {
+            steps {
+                script{
+                    util.spaceLeft()
+                }
+            }
+        }
+        stage('Build projects') {
+            tools {
+              jdk build_jdk_tool
+              maven build_maven_tool
+            }
+            steps {
+                script {
+                    def buildChainActionInfo =
+                      isFDB() ? [action: 'fd', file: 'pull-request-config.yaml'] :
+                      isSingle() ? [action: 'single', file: 'pull-request-config.yaml'] :
+                      [action: 'pr', file: 'pull-request-config.yaml']
+                    withCredentials([string(credentialsId: 'kie-ci1-token', variable: 'GITHUB_TOKEN')]) {
+                        sh "build-chain-action -token=${GITHUB_TOKEN} -df='https://raw.githubusercontent.com/\${GROUP}/kogito-pipelines/master/.ci/${buildChainActionInfo.file}' -folder='bc' build ${buildChainActionInfo.action} -url=${env.ghprbPullLink} --skipParallelCheckout -cct '(^mvn .*)||\$1 -Dmaven.wagon.http.ssl.insecure=true'"
```

Comment:
we should use the `starting-project` option with value from `BUILDCHAIN_PROJECT` if exists

Reply:
@radtriste I will explain how this thing is treated by the build chain tool.
Based on build type (see https://github.com/kiegroup/github-action-build-chain#execution-build-action)
- `pr` and `single` : the starting project is taken from the  `-url` pull request event.
- `fdb`: the chain is composed base on the project triggering the job (the one from `-url` pull request event) and the whole list of projects are built from the bottom to the top of the tree

Reply:
it is as we will call that jenkinsfile for different project, that is the thing about having multiple jobs for the PR for different project.
For example, with Optaplanner, we will have many different PR jobs running in parallel:
- optaplanner
- optaweb*
- optaplanner-quickstarts
- kogito-apps
- kogito-examples

For each of them, we will be with type `pr` and `starting-project` set.

A bit similar to what has been done in kogito-runtimes GHActions: https://github.com/kiegroup/kogito-runtimes/tree/master/.github/workflows

Reply:
@radtriste done by:
* https://github.com/kiegroup/optaplanner/pull/1440/commits/32af7a72f9ff7da1688b4915a9d6ffaab83a2bc9
* https://github.com/kiegroup/github-action-build-chain/pull/161
* https://github.com/kiegroup/kie-jenkins-scripts/pull/1041

many thanks!